Drop `src/libsecp256k1-config.h`

hebasto opened the pull request:
Cherry-picked the first commit from #1142 and addressed a [comment](https://github.com/bitcoin-core/secp256k1/pull/1142#issuecomment-1295099597).

real-or-random commented on configure.ac:
```diff
@@ -293,10 +293,10 @@ int128_struct)
   AC_DEFINE(USE_FORCE_WIDEMUL_INT128_STRUCT, 1, [Define this symbol to force the use of the structure for simulating (unsigned) int128 based wide multiplication])
```
This newly added `AC_DEFINE` should be changed, too.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/secp256k1/pull/1178#issuecomment-1353039879).

hebasto commented:
Updated e129a39bf83eb05911b2c336feaff1580693127e -> ad8647f548cba9c367f837fcdb76de280fa93c19 ([pr1178.01](https://github.com/hebasto/secp256k1/commits/pr1178.01) -> [pr1178.02](https://github.com/hebasto/secp256k1/commits/pr1178.02), [diff](https://github.com/hebasto/secp256k1/compare/pr1178.01..pr1178.02)):

- addressed @real-or-random's [comment](https://github.com/bitcoin-core/secp256k1/pull/1178#discussion_r1048802492)

hebasto commented:
Not sure about this `AC_DEFINE` macro https://github.com/bitcoin-core/secp256k1/blob/9a8d65f07f171b07bd7a33828dce073d819fbdef/build-aux/m4/bitcoin_secp.m4#L23

real-or-random commented:
> Not sure about this `AC_DEFINE` macro
> 
> https://github.com/bitcoin-core/secp256k1/blob/9a8d65f07f171b07bd7a33828dce073d819fbdef/build-aux/m4/bitcoin_secp.m4#L23

I suggest removing it, it's unused, and also confusing because we use the `VALGRIND` macro.

hebasto commented:
> I suggest removing it, it's unused, and also confusing because we use the `VALGRIND` macro.

Done.

real-or-random reviewed (approved):
utACK 9c5a4d21bbe820173e361211f42d5fee4765944c

real-or-random commented on Makefile.am:
```diff
@@ -148,7 +148,7 @@ endif
 noinst_PROGRAMS += ecdsa_example
 ecdsa_example_SOURCES = examples/ecdsa.c
-ecdsa_example_CPPFLAGS = -I$(top_srcdir)/include
+ecdsa_example_CPPFLAGS = -I$(top_srcdir)/include $(SECP_CONFIG_DEFINES)
```
Shouldn't these be dropped for the example, because a user application shouldn't need our defines?

hebasto replied:
See #1185.
